src/test/java/com/softserve/edu/greencity/api/tests/signin/SignInTest.java

```diff
@@ -1,41 +1,97 @@
     public Object[][] invalidCredentials() {
         return new Object[][]{
-                {"", "", "must not be blank"}, //GC-490
-                {"test#test.ua", "Test1!", "The email is invalid"}, //GC-493
-                {"green.city.test10@gmail.com", "Green.city#120", "Bad email or password"}, //Also GC-493
+                {"green.city.test10@gmail.com", "Green.city#120", "The user does not exist by this email"}, //GC-493, GC-507
                 {"green.city.test2@gmail.com", "123456", "Bad password"}, //GC-495
+                {"green.city.test2@gmail.com", "Gr.city#100", "Bad password"} //GC-496
         };
     }
 
-    @Test(dataProvider = "invalidCredentials")
-    @Description("Verifies proper status codes and error messages on providing improper email/password")
-    public void badCredentialsTest(String email, String password, String message) {
+    @DataProvider(name = "validCredentials")
+    public Object[][] validCredentials() { //GC-506
+        return new Object[][]{
+                {"green.city.test2@gmail.com", "Green.city#500"}
+        };
+    }
+
+    /**
+     * Does similar logic for credentials tests.
+     * Also asserts that content type is JSON
+     */
+    private ArrayAssertion getCommonCredentialsAssertion(String email, String password) {
         SignInDto credentials = SignInBuilder.signInWith()
                 .email(email)
                 .password(password)
                 .build();
         OwnSecurityClient ownSecurityClient = new OwnSecurityClient("JSON");
         Response response = ownSecurityClient.signIn(credentials);
         response.print();
-        ArrayAssertion assertion = new ArrayAssertion(response);
+        return new ArrayAssertion(response)
+                .contentType(ContentType.JSON);
+    }
+
+    @Test(dataProvider = "blankCredentials", testName = "GC-490")
+    @Description("Verify that user gets error message when sign in with empty text fields")
+    public void blankCredentialsTest(String email, String password, String message) {
+        ArrayAssertion assertion = getCommonCredentialsAssertion(email, password);
         assertion.statusCode(400)
-                .contentType(ContentType.JSON)
-                .bodyArrayEquals("message", message);
+                .bodyArrayContains("message", message);
+    }
+
+    @Test(dataProvider = "invalidEmail", testName = "GC-493, GC-494")
+    @Description("Verify that user gets error message and no token when sign in with incorrect email")
+    public void invalidEmailTest(String email, String password, String message) {
+        ArrayAssertion assertion = getCommonCredentialsAssertion(email, password);
+        assertion.statusCode(400)
+                .bodyArrayContains("message", message) //For some reason, they return an array here
+                .bodyNoParameter("accessToken");
+    }
+
+
+    @Test(dataProvider = "invalidCredentials", testName = "GC-495, GC-496, GC-507")
```

Comment:
Is it ok to have few tests in one method ?

Reply:
Since the test logic is the same, and test cases really differ only by provided data, it seems to be OK.